### ISSUE TITLE
Spring-security-oauth2.xsd is not published to http://www.springframework.org/schema/security

### ISSUE THREAD

**Issue by lukasondrak96** [spring-projects/spring-security#11357]
**Summary**
Spring-security-oauth2.xsd should be published under https://www.springframework.org/schema/security/
Or was this removal intentional? Because it crashes our application based on spring.

**Actual Behavior** 
Spring-security-oauth2.xsd is not available under https://www.springframework.org/schema/security/ causing XML schema validation failures in IDEs. 

**Expected Behavior**
Spring-security-oauth2.xsd should be available under https://www.springframework.org/schema/security/ to enable successful XML schema validation in IDEs

**Comment by lukasondrak96** [comment]
For anyone missing this file, temporal solution could be something like this:
https://github.com/dhis2/dhis2-core/pull/5097/commits/c1a0f4798d11c6cb90e76a9b3777cf479817321c

**Comment by jgrandja** [comment]
@lukasondrak96 The Spring Security OAuth project has reached end-of-life. Please see [announcement](https://spring.io/blog/2022/06/01/spring-security-oauth-reaches-end-of-life).

As part of the end-of-life process, there are a few cleanup tasks involving the removal of some content, archives, moving repository to attic, etc.

**Comment by bilaliqball** [comment]
@lukasondrak96 
https://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd
also returns 404 , page not found

**Comment by whaan** [comment]
in the [spring.schemas](https://github.com/spring-attic/spring-security-oauth/blob/2.3.x/spring-security-oauth2/src/main/resources/META-INF/spring.schemas) file in the META-INF folder, all requests for `https\://www.springframework.org/schema/security/spring-security-oauth2.xsd` get redirected to  `org/springframework/security/oauth2/spring-security-oauth2-2.0.xsd`

![Screen Shot 2022-06-10 at 13 12 04](https://user-images.githubusercontent.com/77047327/173053109-d12a22d0-e0fa-47b2-9c41-3eadb418fd8e.png)

So the XSD doesn't have to looked up from the internet. As long as you use https, it will simply look it up from the classpath

**Comment by MikeN123** [comment]
Yes, and the problem is that previously `http` was used, other projects consistently added both `http` and `https` to the `spring.schemas`, but that was never fixed for spring-security-oauth. So that's why stuff all over the place is breaking now.

**Comment by MikeN123** [comment]
@jgrandja  I just can't understand the decision to actively start breaking existing or legacy projects. What's next, removing the artifacts from Maven Central? The [end-of-life announcement](https://spring.io/blog/2020/05/07/end-of-life-for-spring-security-oauth) stated that the project would be opened up to the community, this goes 100% against that.

**Comment by MikeN123** [comment]
@jgrandja The point is that I never expected XSD's to be deleted. The Spring Security 2.0 XSD's are still around as well, and that version is certainly end of life as well. 

There is a large gap between 'end of life, no longer maintained' and 'we just instantly disable any and all applications that still use the app and have never switched the XSD do HTTPS as that was never asked'.

**Comment by jgrandja** [comment]
@MikeN123 

> @jgrandja The point is that I never expected XSD's to be deleted. The Spring Security 2.0 XSD's are still around as well, and that version is certainly end of life as well.

Point taken. We'll put back the XSD.

**Comment by MikeN123** [comment]
Thank you very much, saves me from a lot of overtime during the weekend (and yes, we do have a plan to get rid of the dependency and will certainly fix any issues ourselves in the meantime if necessary).

**Comment by jgrandja** [comment]
@MikeN123 I certainly don't want you working overtime. We're on it now. Stay tuned.

**Comment by jzheaux** [comment]
@MikeN123, @lukasondrak96 I've reuploaded the XSD and now https://www.springframework.org/schema/security/spring-security-oauth2.xsd and the HTTP equivalent both resolve for me.

Is there anything else needed to get you up and going again?

**Comment by MikeN123** [comment]
Many thanks, should be fine now :-)

**Comment by jgrandja** [comment]
Reopening as I think this issue is only temporarily resolved as it may become an issue again after the next Spring Security release. We need to investigate further.

cc/ @jzheaux @rwinch

**Comment by jzheaux** [comment]
@jgrandja I don't think this will be an issue. The schema upload is an additive process. That is, the next version's XSD is uploaded to the docs server on each release and added to (not replacing) the existing set of XSDs. Is there something else you'd like to see before closing this issue?

**Comment by pnk0807** [comment]
@jzheaux , @jgrandja 
I would like to inform that the issue occured again. Is it possible to make 
https://www.springframework.org/schema/security/spring-security-oauth2.xsd available again?

**Comment by anihovsk** [comment]
@jzheaux , @jgrandja - hi guys, it would be great if you could upload the xsd back. I assume many applications around the world just cannot be started/restarted due to the issue. It is especially critical for applications in Production which cannot be restarted for maintenance. 
Thanks

**Comment by pnk0807** [comment]
@anihovsk or any affected devs
You can change following line in xml configuration:
from: https://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd
to: https://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd

It will allow during build to use dependency locally stored in jar files.

**Comment by mirihernandez-wahbe** [comment]
Seems the issue is not resolved, actually there is not spring-security-oauth2-2.0.xsd in www.springframework.org/schema/security (http netiher https). Seems all files were renamed as spring-security[XX].xsd as of 02/20/2023 ... none of them has auth2 tags. Any idea to pull data from local env 



Tried with below options but didnt work
xsi:schemaLocation="http://www.springframework.org/schema/security/oauth2  jar:file://C:/myjar/spring-security-oauth2-2.5.2.RELEASE.jar!/org/springframework/security/oauth/spring-security-oauth2/spring-security-oauth2-2.0.xsd"

xsi:schemaLocation="http://www.springframework.org/schema/security/oauth2  
file://C:/myjar/org/springframework/security/oauth/spring-security-oauth2/spring-security-oauth2-2.0.xsd"

Regards

**Comment by MatthewGreigPrecor** [comment]
> @anihovsk or any affected devs You can change following line in xml configuration: from: http://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd to: https://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd So basically change http to https. Works the same as changing to classpath It will allow during build to use dependency locally stored in jar files.

Doesn't help, we still get the 404 and were already using https

**Comment by mirihernandez-wahbe** [comment]
The way it got fixed is by opening the jar (spring-security-oauth2-XXX) file with winzip 
then go to org\springframework\security you will find xsds name

now point to https://www.springframework.org/schema/security/XSDFileNameInsideJAR.xsd
Hope it helps

**Comment by norman-knott** [comment]
The `http to https` trick worked for me. Just wondering if the removal of the XSD happened intentionally this time.

**Comment by sjohnr** [comment]
@norman-knott It was not intentional, but is unfortunately a gap in our release process (when combined with an [automatic process](https://github.com/spring-projects/spring-security/issues/11365) on the backend to point to the latest of our deployed schemas). I've opened gh-12804 to address the specific issue and will be submitting a fix soon so that subsequent releases don't re-trigger the issue.

This issue has been addressed in the meantime with manual intervention, so I'm going to close this issue in favor of gh-12804.
